R/utils-json.R

```diff
@@ -278,10 +281,51 @@ subsetByMonth <- function(pts, onlyMonth) {
   }
   return(pts)
 }
-getYvals_approvals <- function(object, num_vals){
-  ylim <- ylim(object)$side.2[1]
-  yvals <- rep(ylim, num_vals)
-  return(yvals)
+
+ApprovalInfoY <- function(object, times, c) {
+  # Compute top and bottom vertical position of approval bars.
+  # 
+  # Args:
+  #   object: A gsplot object.
+  #   times: "times" argument of R rep() function.
+  #   c: A scaling ratio to adjust top or bottom of approval bar rectangle.
+  #
+  # Returns:
+  #   Approval bar, top or bottom y-axis point, in world coordinates.
+  reverse <- object$side.2$reverse
+  
+  # no idea whether semantics of object$global$par$ylog are correct in this
+  # context or not; it just looked promising in the debugger
+  if (is.null(object$global$par$ylog)) {
```

Comment:
Right, but I wonder if we would like it to terminate with an error if it comes through as NULL. That might flag us that something else is going wrong. We can always change that later though